tests/test_datasets/test_hierarchical_dataset.py

```diff
@@ -0,0 +1,169 @@
+import pandas as pd
+import pytest
+
+from etna.datasets.hierarchical_structure import HierarchicalStructure
+from etna.datasets.tsdataset import TSDataset
+
+
+@pytest.fixture
+def hierarchical_structure():
+    hs = HierarchicalStructure(
+        level_structure={"total": ["X", "Y"], "X": ["a", "b"], "Y": ["c", "d"]},
+        level_names=["total", "market", "product"],
+    )
+    return hs
+
+
+@pytest.fixture
+def diff_level_segments_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 2,
+            "segment": ["X"] * 2 + ["a"] * 2,
+            "target": [1, 2] + [10, 20],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def diff_level_segments_df_exog():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 2,
+            "segment": ["X"] * 2 + ["a"] * 2,
+            "exog": [1, 2] + [10, 20],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def missing_segments_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"],
+            "segment": ["X"] * 2,
+            "target": [1, 2],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def market_level_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 2,
+            "segment": ["X"] * 2 + ["Y"] * 2,
+            "target": [1, 2] + [10, 20],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def market_level_df_exog():
+    df_exog = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 2,
+            "segment": ["X"] * 2 + ["Y"] * 2,
+            "exog": [1, 2] + [10, 20],
+        }
+    )
+    df_exog = TSDataset.to_dataset(df_exog)
+    return df_exog
+
+
+@pytest.fixture
+def product_level_df():
+    df = pd.DataFrame(
+        {
+            "timestamp": ["2000-01-01", "2000-01-02"] * 4,
+            "segment": ["a"] * 2 + ["b"] * 2 + ["c"] * 2 + ["d"] * 2,
+            "target": [1, 2] + [10, 20] + [100, 200] + [1000, 2000],
+        }
+    )
+    df = TSDataset.to_dataset(df)
+    return df
+
+
+@pytest.fixture
+def simple_hierarchical_ts(market_level_df, hierarchical_structure):
+    df = market_level_df
```

Comment:
Do we need this line?

Reply:
Not at all, but i think it makes code readable, as you replace long named fixtures with common `df`, `df_exog`. And it is still reflects the meaning of the fixtures as we initialize `df`, `df_exog` in the begining